wpiutil/src/main/java/edu/wpi/first/util/FileLogger.java

```diff
@@ -0,0 +1,28 @@
+package edu.wpi.first.util;
+
+import edu.wpi.first.util.datalog.DataLog;
+
+/**
+ * A class version of `tail -f`, otherwise known as `tail -f` at home. Watches a file and puts the
+ * data into a data log. Only works on Linux-based platforms.
+ */
+public class FileLogger implements AutoCloseable {
+  private long m_impl;
```

Comment:
PMD is emitting an [`ImmutableField`](https://pmd.github.io/pmd/pmd_rules_java_design.html#immutablefield) warning.